src/components/AttachmentCarousel/index.js

```diff
@@ -242,7 +242,10 @@ class AttachmentCarousel extends React.Component {
      * @returns {JSX.Element}
      */
     renderCell(props) {
-        const style = [props.style, styles.h100, {width: this.state.containerWidth}];
+        // Use window width instead of layout width to address the issue in https://github.com/Expensify/App/issues/17760
+        // considering horizontal margin and border width in centered modal
+        const modalStyles = styles.centeredModalStyles(this.props.isSmallScreenWidth);
+        const style = [props.style, styles.h100, {width: this.props.windowWidth - (modalStyles.marginHorizontal + modalStyles.borderWidth) * 2 + 1}];
```

Comment:
Thanks for the ping @aimane-chnaif 

As a developer, I have concerns with this approach as it seems to create an implicit coupling between components.

If, for any reason, someone changes the generic modal dimensions through methods other than `marginHorizontal` or `borderWidth`, they could inadvertently disrupt the `width` utilized here without even realizing it. 

Moreover, the Carousel appears to be aware of its rendering context within a centered modal - a detail which, ideally, it should not be concerned with.

It seems we might be trading one bug, that only occurs on resize, for another which could potentially occur all the time.

Reply:
If addressing this bug is crucial, my preference would be to continue using `containerWidth`, but dynamically adjust its value in accordance with changes in window size. Here's a step-by-step outline of how this could work:

1. Upon opening the carousel, we capture the initial values of both `containerWidth` and `windowWidth`.
2. Changes to the window size are broadcast first, allowing us to respond instantly as these changes occur.
3. We then calculate the proportion by which `windowWidth` has either shrunk or expanded, and immediately apply this same proportion to `containerWidth`.

This solution is universal and does not depend on specifics. It remains effective regardless of specific properties or the modal context, thereby reducing the risk of inadvertently introducing new issues.